src/pani/async.cljs

```diff
@@ -0,0 +1,85 @@
+(ns pani.async
+  (:require [pani.core :as p]
+            [cljs.core.async :refer [<! >! chan put! close!]])
+  (:require-macros [cljs.core.async.macros :refer [go go-loop]]))
+
+;; (defn chan-val [] (chan (map p/value)))
+
+;; (defn chan-snap [] (chan (map p/wrap-snapshot)))
+
+(defn with-chan "Dumb pseudo-macro" [f]
+  (let [ch (chan)]
+    (f ch)
+    ch))
+
+(defn chan->cb
+  "Create callback that pushes arguments onto given chan, and closes on nil"
+  [ch]
+  (fn [val]
+    (if val
+      (put! ch val)
+      (close! ch))))
+
+(defn chan->cb-once
+  "Create callback that pushes arguments onto chan exactly once"
+  [ch]
+  (fn [val]
+    (if val
+      (put! ch val))
```

Comment:
Good spot, but think will stick with how it is:

1. Would need to close in the other branch anyway, prefer one-size-fits-all
2. No reason to wait until value consumed then close (3rd param is callback), value will still be delivered if it closes the door on its way in.